tests/v1/core/test_async_scheduler.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2025 Rebellions Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at:
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+import pytest
+from vllm.v1.core.sched.output import SchedulerOutput
+from vllm.v1.outputs import ModelRunnerOutput
+
+from .utils import create_requests, create_scheduler
+
+
+def _make_model_runner_output(
+    scheduler_output: SchedulerOutput, ) -> ModelRunnerOutput:
+    req_ids = list(scheduler_output.num_scheduled_tokens.keys())
+    return ModelRunnerOutput(
+        req_ids=req_ids,
+        req_id_to_index={
+            req_id: i
+            for i, req_id in enumerate(req_ids)
+        },
+        sampled_token_ids=[[i] for i in range(len(req_ids))],
+        spec_token_ids=None,
+        logprobs=None,
+        prompt_logprobs_dict={},
+    )
+
+
+@pytest.mark.parametrize(
+    "max_num_seqs, block_size, max_model_len, "
+    "num_blocks, num_tokens_per_batch, "
+    "exp_new_req0_blocks, exp_new_req1_blocks, "
+    "exp_cached0_new, exp_cached1_new",
+    [
+        pytest.param(
+            2, 16, 64, 8, 32, [1, 2], [3, 4], [5], [6], id="16bsize-32len"),
+        pytest.param(
+            2, 8, 24, 7, 17, [1, 2, 3], [4, 5, 6], [], [], id="8bsize-17len")
```

Comment:
```suggestion
            2, 8, 24, 6, 17, [1, 2, 3], [4, 5, 6], [], [], id="8bsize-17len")
```
`max_num_seqs`(2) * (`max_model_len`(24) // `block_size`(8)) = 6?

Reply:
1. We need null block.
2. I thought excessive block is not a problem.

Reply:
Oh I got it. Thanks!